src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1276,6 +1282,43 @@ protected override void ProcessRecord()
                     WriteObject(res);
                 }
             }
+            else if (OnlyMatching)
+            {
+                string line;
+                if (_inputObject.BaseObject is MatchInfo matchInfo)
+                {
+                    line = matchInfo.Line;
+                }
+                else 
+                {
+                    line = LanguagePrimitives.ConvertTo<string>(_inputObject.BaseObject);
+                }
+                
+                foreach (Regex regexValue in _regexPattern)
+                {
+                    if (AllMatches)
+                    {
+                        MatchCollection mc = regexValue.Matches(line);
+                        if (mc.Count <= 0)
+                        {
+                            continue;
+                        }
+
+                        foreach (Match matchObject in mc)
+                        {
+                            WriteObject(matchObject.Value);
```

Comment:
```suggestion
                            WriteObject(matchObject.Value, true);
```

Please ensure you use the overload for WriteObject that enumerates the output, as almost all cmdlets including `Select-String` currently enumerate their output.